graphqlapi/src/main/java/io/stargate/graphql/PlaygroundServlet.java

```diff
@@ -15,563 +15,57 @@
  */
 package io.stargate.graphql;
 
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.stream.Collectors;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.osgi.framework.Bundle;
 
 public class PlaygroundServlet extends HttpServlet {
+
+  private final String playgroundFile;
+
+  public PlaygroundServlet(Bundle bundle) throws IOException {
+    // From
+    // https://raw.githubusercontent.com/prisma-labs/graphql-playground/master/packages/graphql-playground-html/withAnimation.html
+    URL entry = bundle.getEntry("/playground.html");
+    // Save the templated file away for later so that we only have to do this conversion once.
+    playgroundFile =
+        new BufferedReader(new InputStreamReader(entry.openConnection().getInputStream()))
+            .lines()
+            .collect(Collectors.joining("\n"));
+  }
+
   @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response) {
     response.setStatus(HttpServletResponse.SC_OK);
     response.setContentType("text/html");
     response.setCharacterEncoding("utf-8");
-    // From
-    // https://raw.githubusercontent.com/prisma-labs/graphql-playground/master/packages/graphql-playground-html/withAnimation.html
+
+    String token = request.getHeader("x-cassandra-token");
+
+    // Replace the templated text with the token if it exist. Using java.lang.String.replaceFirst
+    // since it's safer than java.lang.String.format(java.lang.String, java.lang.Object...) due to
+    // the percent signs that exist in the string.
+    String formattedPlaygroundFile =
+        playgroundFile.replaceFirst("AUTHENTICATION_TOKEN", token == null ? "" : token);
```

Comment:
NIT: We could use marker delimitation like `{{AUTHENTICATION_TOKEN}}` or `?AUTHENTICATION_TOKEN?` to avoid unintended replacements.